File: benchmark/results/ts-monolith-opus-script/run-002/b-baseline/output/src/notifications/services/DeliveryDispatcher.ts
import { EmailTemplate } from "../value-objects/EmailTemplate";
import { Id } from "../../shared/value-objects/Id";

/** Dispatches notifications to delivery channels. */
export class DeliveryDispatcher {
  create(input: Partial<EmailTemplate>): EmailTemplate {
    void input;
    throw new Error("not implemented");
  }

  findById(id: Id): EmailTemplate | null {
    void id;
    throw new Error("not implemented");
  }

  list(limit: number): EmailTemplate[] {
    void limit;
    throw new Error("not implemented");
  }

  delete(id: Id): void {
    void id;
    throw new Error("not implemented");
  }
}

File: benchmark/results/python-multilang/run-002/b-baseline/output/metaengine_demo/notification/services/delivery_dispatcher.py
"""DeliveryDispatcher for the notification domain."""

from typing import Optional

from metaengine_demo.notification.aggregates.notification import Notification


class DeliveryDispatcher:
    """Service responsible for dispatching notifications to channels."""

    def create(self, input: Optional[Notification]) -> Notification:
        """Dispatch a newly created notification."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Notification]:
        """Look up dispatch state for a notification by id."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Notification]:
        """List dispatched notifications up to the given limit."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Cancel dispatch for a notification by id."""
        raise NotImplementedError("not implemented")
